sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -984,7 +984,16 @@ object SparkSession extends Logging {
    *
    * @since 2.0.0
    */
+  @deprecated("This method is deprecated and will be removed in future versions.", "3.1.0")
   def setActiveSession(session: SparkSession): Unit = {
+    if (SQLConf.get.legacyAllowModifyActiveSession) {
+      setActiveSessionInternal(session)
+    } else {
+      throw new UnsupportedOperationException("Not allowed to modify active Spark session.")
```

Comment:
Can we add the flag in the error message? Then users will know how to unblock themselves without googling.

Reply:
+1. @leanken can you create a follow-up? thanks!

Reply:
OK.